src/functionNameRule.ts

```diff
@@ -128,25 +128,50 @@ function walk(ctx: Lint.WalkContext<Options>) {
     function cb(node: ts.Node): void {
         if (tsutils.isMethodDeclaration(node)) {
```

Comment:
These `if` blocks are getting pretty huge, and have a lot of variables declared in them. This rule's ever-increasing complexity is likely a partial cause of the option being overlooked before - it's really hard to keep track of the logic here!

Let's split these blocks into separate functions.

Reply:
Thanks for the feedback.  I was trying to do minimal changes, maintaining the existing logical flow and structure until a cleaner rework of these options came about.

Anyway, I have updated the code to break the validations out into separate functions and have different handlers for each configuration.  I am open to changing it further with more specific instructions.